crates/red_knot_python_semantic/resources/mdtest/binary/instances.md

```diff
@@ -263,7 +260,10 @@ class B:
     __add__ = A()
 
 # TODO: this could be `int` if we declare `B.__add__` using a `Callable` type
-reveal_type(B() + B())  # revealed: Unknown | int
+# TODO: This should not be an error but we currently fail to bind `self` in `A.__call__` as `B`
```

Comment:
This is a "regression" in the sense that we need to understand that `__add__ = A` rebinds the `self` in `A.__call__` to `B`.